.github/workflows/lint.yml

```
@@ -36,14 +36,36 @@ jobs:
           echo "ShellCheck version:" && shellcheck --version
-          SH_FILES=$(git ls-files | grep -E '(^bin/git-shiplog$|^contrib/hooks/[^/]+$|\.sh$)' || true)
-          if [ -n "$SH_FILES" ]; then
-            shellcheck -e SC2034 -S style -s bash $SH_FILES
+          if [ "${{ github.event_name }}" = "pull_request" ]; then
+            if [ "${{ steps.changed-shell.outputs.any_changed }}" != "true" ]; then
+              echo "No changed shell files; skipping."
+              exit 0
+            fi
+            files='${{ steps.changed-shell.outputs.all_changed_files }}'
+            echo "Changed shell files (raw):
+$files"
```

actionlint (1.7.7)

could not parse as YAML: yaml: line 59: could not find expected ':'

(syntax-check)